addons/calendar/models/mail_activity_mixin.py

```diff
@@ -16,4 +16,5 @@ def _compute_activity_calendar_event_id(self):
         """This computes the calendar event of the next activity.
         It evaluates to false if there is no such event."""
         for record in self:
-            record.activity_calendar_event_id = fields.first(record.activity_ids).calendar_event_id
+            activities = record.activity_ids
+            record.activity_calendar_event_id = next(iter(activities), activities).calendar_event_id
```

Comment:
I don't like this change, as it reduces code readability.  Most developers don't know what `next()` with two parameters does.  Even I need more time to figure out that code, because it's not idiomatic (yet).  I find it acceptable in the framework's implementation, but not in business code.

I suggest to fix the "prefetching" of recordset slices instead, so one could write
```suggestion
            record.activity_calendar_event_id = record.activity_ids[:1].calendar_event_id
```

Reply:
I made the code clearer (an `if` instead of the default) and it keeps the prefetch.

About the prefetch, it should be discussed separately because it may change behaviour. We can tackle it at the same time we will discuss the prefetch of `filtered`.